### PR TITLE
[PVR/EPG]: debug log now as configurable logging component

### PR DESCRIPTION
**Pull request by mk01**

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -2988,7 +2988,12 @@ msgctxt "#680"
 msgid "Verbose logging for the [B]Video[/B] component"
 msgstr ""
 
-#empty strings from id 681 to 699
+#: xbmc/settings/AdvancedSettings.cpp
+msgctxt "#681"
+msgid "Verbose logging for the [B]PVR[/B] component"
```
Comment (MartijnKaijser) [inline addons/resource.language.en_gb/resources/strings.po]: Please do not use low range text string id

**Comment by ksooo** [comment]
I guess you just copied what others did wrt configurable log components, but eouldn't it make sense to extend CLog::Log with a new parameter "component" and to do the <code>g_advancedSettings.CanLogComponent("component")</code> check only once inside CLog::Log instead of doing the check before each call to CLog::Log IMO, current approach clutters our code base without any real need. Just my 2 ct.

**Comment by Jalle19** [comment]
@ksooo I had a similar concern with tvheadend, the rationale was that doing a check each time a log function is called could be a performance issue. I don't really buy that myself though.

**Comment by ksooo** [comment]
My main concern is cleared code.

**Comment by ksooo** [comment]
I ment cluttered code. ;-)

**Comment by mk01** [comment]
@Jalle19
the check must happen at some point, so from the performance perspective I suppose it is neutral.

@ksooo 
I had idea like this, just didn't want to change interfaces (at least in one PR). if you all agree on such change, we can do it and mass-change all such calls throughout the code.

**Comment by mk01** [comment]
rebase

**Comment by mk01** [comment]
@ksooo @Jalle19 

after some thoughts it seems that existing code provides straight forward solution to our discussion.

what do you think ?

File: addons/resource.language.en_gb/resources/strings.po [hunk 2]
```diff
@@ -4947,7 +4947,12 @@ msgctxt "#10524"
 msgid "Movie information"
 msgstr ""
 
-#empty strings from id 10525 to 11999
+#: xbmc/settings/AdvancedSettings.cpp
+msgctxt "#10525"
+msgid "Verbose logging for the [B]PVR[/B] component"
```
Comment (Razzeee) [inline addons/resource.language.en_gb/resources/strings.po]: Are the other strings also bold? Is that something we do?

Comment (Montellese) [inline addons/resource.language.en_gb/resources/strings.po]: Yes the main "component" is always bold for these strings.

**Comment by mk01** [comment]
@ksooo 
rebase

**Comment by stefansaraev** [comment]
what's the plan here

**Comment by ksooo** [comment]
I like this. +1 from my end.

@mk01 ping. needs rebase

File: xbmc/utils/log.cpp [hunk 3]
```diff
@@ -166,25 +167,24 @@ int CLog::GetLogLevel()
...
   CSingleLock waitLock(s_globals.critSec);
-  s_globals.m_extraLogLevels = level;
+  s_globals.m_extraLogLevels = level & ~LOGMASK;
 }
 
 bool CLog::IsLogLevelLogged(int loglevel)
 {
   const int extras = (loglevel & ~LOGMASK);
-  if (extras != 0 && (s_globals.m_extraLogLevels & extras) == 0)
-    return false;
```
Comment (FernetMenta) [inline xbmc/utils/log.cpp]: this changes behaviour. compiling with _DEBUG would result in logging all.

Comment (mk01) [inline xbmc/utils/log.cpp]: @FernetMenta 
right, thanks. should be fixed now with 76213b1d611c06d798015529a90b100739a85b34

**Comment by mk01** [comment]
@ksooo 
rebased

**Comment by Rechi** [comment]
PR has a merge commit

**Comment by ksooo** [comment]
Please rebase your branch against master, do not merge master into your branch.

**Comment by mk01** [comment]
guys, for almost a year I was away and busy. maybe so much didn't change on Logging.

**Comment by ksooo** [comment]
@mk01 Please rebase your branch against master, do not merge master into your branch.

**Comment by ksooo** [comment]
@mk01 ping ping ping, if you would rebase, we could merge it. Anything unclear whit what we mean wrt the merge commit?

**Comment by MartijnKaijser** [comment]
replaced by https://github.com/xbmc/xbmc/pull/14112

**Pull request closed without merging**